### ISSUE TITLE
1.7 fails to start

### ISSUE THREAD

**Issue by Athanasius** [jixxed/ed-odyssey-materials-helper#2]
I had a browser tab open ready to install this and it still showed 1.6 as latest this morning (i.e. browser hadn't refreshed the tab) and that seemed to work fine.

Then I checked the settings and it pointed out that 1.7 was available.

1. Download 1.7 .msi
2. Install it over the top
3. Nothing happens when run
4. Uninstall via Windows Settings > Apps
5. Install 1.7 again, this time 'clean'
6. Still doesn't run

Is there any logging that can help with this?

OS: Windows 10 Home 20H2, with June 2021 updates.
Plenty of disk space free, and not using much of my 32GB RAM right now.

**Comment by Athanasius** [comment]
Uninstalling 1.7 and installing 1.6 this seems to be a valid way to run it in cmd.exe:

`\AppData\Local\Elite Dangerous Odyssey Materials Helper>"runtime\bin\Elite Dangerous Odyssey Materials Helper.bat"`

But doing that with 1.7:

```
AppData\Local\Elite Dangerous Odyssey Materials Helper>"runtime\bin\Elite Dangerous Odyssey Materials Helper.bat"
Exception in Application start method
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplicationWithArgs(Unknown Source)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication(Unknown Source)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.base/java.lang.reflect.Method.invoke(Unknown Source)
        at java.base/sun.launcher.LauncherHelper$FXHelper.main(Unknown Source)
Caused by: java.lang.RuntimeException: Exception in Application start method
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(Unknown Source)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.util.NoSuchElementException: No line found
        at java.base/java.util.Scanner.nextLine(Unknown Source)
        at nl.jixxed.eliteodysseymaterials@1.6/nl.jixxed.eliteodysseymaterials.watchdog.JournalWatcher.isOdysseyJournal(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(Unknown Source)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
        at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.reduce(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline.max(Unknown Source)
        at nl.jixxed.eliteodysseymaterials@1.6/nl.jixxed.eliteodysseymaterials.watchdog.JournalWatcher.findLatestFile(Unknown Source)
        at nl.jixxed.eliteodysseymaterials@1.6/nl.jixxed.eliteodysseymaterials.watchdog.JournalWatcher.watch(Unknown Source)
        at nl.jixxed.eliteodysseymaterials@1.6/nl.jixxed.eliteodysseymaterials.Main.start(Unknown Source)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(Unknown Source)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(Unknown Source)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
        at java.base/java.security.AccessController.doPrivileged(Unknown Source)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source)
        at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
        at javafx.graphics/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at javafx.graphics/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
        ... 1 more
Exception running application nl.jixxed.eliteodysseymaterials.Main
```

**Comment by jixxed** [comment]
When the game starts it creates an empty journal file. a bit later it writes a fileheader to the log that it is an odyssey game.
the app detects the creation of the file, but expects the first line to be there. 
will fix this for the next version
as a workaround you can login to elite and then start the app.

**Comment by jixxed** [comment]
fixed in 1.8.  aforementioned workaround also does not work as there was another bug in there. 
pulled the 1.7 release

**Comment by Athanasius** [comment]
1.8 confirmed working here, thanks.
